packages/squarev2-integration/src/SquareV2PaymentMethod.spec.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+    CheckoutSelectors,
+    CheckoutService,
+    createCheckoutService,
+    LanguageService,
+    PaymentInitializeOptions,
+    PaymentRequestOptions,
+} from '@bigcommerce/checkout-sdk';
+import { mount } from 'enzyme';
+import React, { FunctionComponent } from 'react';
+import { act } from 'react-dom/test-utils';
+
+import {
+    PaymentFormService,
+    PaymentMethodProps,
+} from '@bigcommerce/checkout/payment-integration-api';
+import { LoadingOverlay } from '@bigcommerce/checkout/ui';
+
+import { getSquareV2 } from './mocks/squarev2-method.mock';
+import SquareV2PaymentMethod from './SquareV2PaymentMethod';
+
+describe('SquareV2 payment method', () => {
+    let checkoutService: CheckoutService;
+    let initializePayment: jest.SpyInstance<
+        Promise<CheckoutSelectors>,
+        [options: PaymentInitializeOptions]
+    >;
+    let deinitializePayment: jest.SpyInstance<
+        Promise<CheckoutSelectors>,
+        [options: PaymentRequestOptions]
+    >;
+    let checkoutState: CheckoutSelectors;
+    let props: PaymentMethodProps;
+    let SquareV2PaymentMethodTest: FunctionComponent;
+
+    beforeEach(() => {
+        checkoutService = createCheckoutService();
+        initializePayment = jest
+            .spyOn(checkoutService, 'initializePayment')
+            .mockResolvedValue(checkoutState);
+        deinitializePayment = jest
+            .spyOn(checkoutService, 'deinitializePayment')
+            .mockResolvedValue(checkoutState);
+        checkoutState = checkoutService.getState();
+        props = {
+            method: getSquareV2(),
+            checkoutService,
+            checkoutState,
+            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
Can we add this rules in `eslintrc` file for spec files, so we have a central place of noting overrides. And therefore clean them up once we upgrade jest.